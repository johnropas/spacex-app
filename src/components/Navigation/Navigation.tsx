import React from 'react';
import './Navigation.scss';
import { NavLink } from 'react-router-dom';
import CONSTANTS from '../../helpers/constants';

const Navigation = () => {
  return (
    <div className="navigation">
      <div className="navbar">
        <div className="navbar-container container">
          <NavLink exact={true} to={CONSTANTS.APP_ROUTES.DASHBOARD} className="navbar-logo">
          </NavLink>
          <div className="menu">
            <ul className="menu-list">
              <NavLink exact={true} to={CONSTANTS.APP_ROUTES.DASHBOARD} className="menu-link">
                <li className="menu-item">
                  <div className="menu-item-wrapper">
                    <span className="menu-text">DASHBOARD</span>
                  </div>
                </li>
              </NavLink>
              <NavLink exact={true} to={CONSTANTS.APP_ROUTES.LOGIN} className="menu-link">
                <li className="menu-item">
                  <div className="menu-item-wrapper">
                    <span className="menu-text">LOGOUT</span>
                  </div>
                </li>
              </NavLink>
            </ul>
          </div>
        </div>
      </div>
    </div>
  );
};

export default Navigation;
