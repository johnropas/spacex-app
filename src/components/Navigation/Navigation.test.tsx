import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Navigation from './Navigation';

it('Instantiates Navigation Component', () => {
  const component = renderer.create(<Router><Navigation /></Router>);
  expect(component).toBeTruthy();
});

it('Renders Navigation Component', () => {
  const component = renderer.create(<Router><Navigation /></Router>);
  expect(component).toMatchSnapshot();
});
