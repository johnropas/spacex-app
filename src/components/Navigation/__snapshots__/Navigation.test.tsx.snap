// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Navigation Component 1`] = `
<div
  className="navigation"
>
  <div
    className="navbar"
  >
    <div
      className="navbar-container container"
    >
      <a
        aria-current="page"
        className="navbar-logo active"
        href="/"
        onClick={[Function]}
        style={Object {}}
      />
      <div
        className="menu"
      >
        <ul
          className="menu-list"
        >
          <a
            aria-current="page"
            className="menu-link active"
            href="/"
            onClick={[Function]}
            style={Object {}}
          >
            <li
              className="menu-item"
            >
              <div
                className="menu-item-wrapper"
              >
                <span
                  className="menu-text"
                >
                  HOME
                </span>
              </div>
            </li>
          </a>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
